plugins/outputs/signalfx/parse/parse.go

```diff
@@ -0,0 +1,58 @@
+package parse
+
+import (
+	"fmt"
+)
+
+// RemoveSFXDimensions removes dimensions used only to identify special metrics for SignalFx
+func RemoveSFXDimensions(metricDims map[string]string) {
+	// remove the sf_metric dimension
+	delete(metricDims, "sf_metric")
+	// remove sf_prefix if it exists in the dimension map
+	delete(metricDims, "sf_prefix")
+}
+
+// SetPluginDimension sets the plugin dimension to the metric name if it is not already supplied
+func SetPluginDimension(metricName string, metricDims map[string]string) {
+	// If the plugin doesn't define a plugin name use metric.Name()
+	if _, in := metricDims["plugin"]; !in {
+		metricDims["plugin"] = metricName
+	}
+}
+
+// GetMetricName combines telegraf fields and tags into a full metric name
+func GetMetricName(metric string, field string, dims map[string]string) (name string, isSFX bool) {
+	// If sf_metric is provided
+	if name, isSFX = dims["sf_metric"]; isSFX {
+		return
+	}
+
+	// If sf_prefix is provided use it instead of metric name
+	if name, isSFX = dims["sf_prefix"]; !isSFX {
+		name = metric
```

Comment:
This is very confusing assigning the name in the line above and then reassigning it here.